dask_sql/physical/rex/core/call.py

```diff
@@ -416,6 +422,48 @@ def extract(self, what, df: SeriesOrScalar):
             raise NotImplementedError(f"Extraction of {what} is not (yet) implemented.")
 
 
+class CeilFloorOperation(PredicteBasedOperation):
+    """
+    Apply ceil/floor operations on a series depending on its dtype (datetime like vs normal)
+    """
+
+    def __init__(self, round_method: str):
+        assert round_method in {
+            "ceil", "floor"
+        }, f"Round method can only be either ceil or floor"
+
+        super().__init__(
+            lambda x: pd.api.types.is_datetime64_any_dtype(x) or isinstance(x, datetime), # if the series is dt type
+            self._round_datetime,
+            getattr(da, round_method)
+        )
+
+        self.round_method = round_method
+
+    def _round_datetime(self, *operands):
+        df, unit = operands
+
+        if is_frame(df):
```

Comment:
As we need this part multiple times (here and e.g. also in the "last_day" function): should we create a utility function?